### PR TITLE
[Task] Add new endpoint to check status of tasks

### LINKED ISSUE #6029: [Task] Add new endpoint to check status of tasks
Hi team,

This issue is related with #5537 and #5387. With the new agent upgrade system, a new task module has been added.

This module is in charge of performing different types of tasks, currently only the new upgrade module takes advantage of it.

In this issue we are looking to create a new endpoint to check the status of the tasks.

## Design

The endpoint should have the following structure:

Path: 

- `GET /tasks/status`

Query parameters: 
- `pretty`
- `wait_for_complete`
- `list_tasks`

## Tasks

- [x] Update spec.yaml
- [x] Add tasks controller
- [x] Add tasks functions at framework side
- [x] Add and run new integration tests
- [x] Add and run new unit tests
- [x] Update changelog

### PR DESCRIPTION
**Pull request by AdriiiPRodri**
|Related issue|
|---|
|#6029|

Hi team,

This PR closes #6029 and it should be merge after the close of this issue #5387. In this PR we have added a new endpoint to check the status of one or more tasks according to its task_id.

### Integration test
```
adriiiprodri@wazuh pytest -vv -x test_tasks_endpoints.tavern.yaml
=========================================================================== test session starts ===========================================================================
platform linux -- Python 3.8.5, pytest-6.1.1, py-1.9.0, pluggy-0.13.1 -- /usr/bin/python3
cachedir: .pytest_cache
rootdir: /home/adriiiprodri/Desktop/git/wazuh/api/test/integration, configfile: pytest.ini
plugins: tavern-1.7.0
collected 2 items                                                                                                                                                         

test_tasks_endpoints.tavern.yaml::PUT /agents/upgrade PASSED                                                                                                        [ 50%]
test_tasks_endpoints.tavern.yaml::GET /tasks/status PASSED                                                                                                          [100%]

================================================================ 2 passed, 3 warnings in 387.07s (0:06:27) ================================================================
```

### Unittest
```
adriiiprodri@wazuh python3 -m pytest -x framework/wazuh/tests/test_tasks.py
=========================================================================== test session starts ===========================================================================
platform linux -- Python 3.8.5, pytest-6.1.1, py-1.9.0, pluggy-0.13.1
rootdir: /home/adriiiprodri/Desktop/git/wazuh/framework
plugins: tavern-1.7.0
collected 2 items                                                                                                                                                         

framework/wazuh/tests/test_tasks.py ..                                                                                                                              [100%]

============================================================================ 2 passed in 0.08s ============================================================================
```

### CODE REVIEW

**Review by davidjiglesias — CHANGES_REQUESTED** [review]
Add documentation PR.
Rebase branch.

File: api/api/spec/spec.yaml [hunk 1]
```diff
@@ -14979,6 +14996,38 @@ paths:
         '429':
           $ref: '#/components/responses/TooManyRequestsResponse'
 
+  /tasks/status:
+    get:
+      tags:
+        - tasks
+      summary: "Get the status of a specified list of tasks"
```
Comment (davidjiglesias) [inline api/api/spec/spec.yaml:15003]:
```suggestion
      summary: "List tasks"
```

File: api/api/spec/spec.yaml [hunk 2]
```diff
@@ -14979,6 +14996,38 @@ paths:
...
+      description: "Returns all available information about the specified list of tasks"
```
Comment (davidjiglesias) [inline api/api/spec/spec.yaml:15004]:
```suggestion
      description: "Returns all available information about the specified tasks"
```

File: api/api/spec/spec.yaml [hunk 3]
```diff
@@ -14979,6 +14996,38 @@ paths:
...
+        - $ref: '#/components/parameters/wait_for_complete'
+        - $ref: '#/components/parameters/tasks_list'
+      responses:
+        '200':
+          description: "Tasks's status"
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ApiResponse'
+              example:
```
Comment (davidjiglesias) [inline api/api/spec/spec.yaml:15019]: Is this example a response from this endpoint?

File: api/test/integration/test_tasks_endpoints.tavern.yaml [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+        force: true
+    response:
+      status_code: 200
+
+---
+test_name: GET /tasks/status
+
+stages:
+
+  - name: Missing task IDs
```
Comment (davidjiglesias) [inline api/test/integration/test_tasks_endpoints.tavern.yaml:29]: Why are we not giving back all tasks if nothing is specified? Does the task socket accept such petition?

File: framework/wazuh/core/tasks.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
+# Copyright (C) 2015-2019, Wazuh Inc.
```
Comment (davidjiglesias) [inline framework/wazuh/core/tasks.py:1]:
```suggestion
# Copyright (C) 2015-2020, Wazuh Inc.
```

File: framework/wazuh/core/tasks.py [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+from wazuh.core.database import Connection
+from wazuh.core.exception import WazuhException, WazuhError, WazuhInternalError, WazuhResourceNotFound
+from wazuh.core.ossec_queue import OssecQueue
+from wazuh.core.utils import chmod_r, WazuhVersion, plain_dict_to_nested_dict, get_fields_to_nest, WazuhDBQuery, \
+    WazuhDBQueryDistinct, WazuhDBQueryGroupBy, SQLiteBackend, WazuhDBBackend, safe_move
+from wazuh.core.wazuh_socket import OssecSocket, OssecSocketJSON
+
+
+def send_to_tasks_socket(command):
+    """Send command to upgrade module / task module
```
Comment (davidjiglesias) [inline framework/wazuh/core/tasks.py:30]:
```suggestion
    """Send command to task module
```

File: framework/wazuh/core/tasks.py [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+
+    Parameters
+    ----------
+    command
+
+    Returns
+    -------
+    Message received from the socket (Task module or Upgrade module)
```
Comment (davidjiglesias) [inline framework/wazuh/core/tasks.py:38]:
```suggestion
    Message received from the socket
```

File: framework/wazuh/core/tasks.py [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+    """
+    # Send upgrading command
```
Comment (davidjiglesias) [inline framework/wazuh/core/tasks.py:40]: ????

**Review by davidjiglesias — CHANGES_REQUESTED** [review]

**Comment by AdriiiPRodri** [comment]
## Unittests
```
adriiiprodri@wazuh pytest wazuh/tests/test_tasks.py
=========================================================================== test session starts ===========================================================================
platform linux -- Python 3.8.6, pytest-6.1.2, py-1.9.0, pluggy-0.13.1
rootdir: /home/adriiiprodri/Desktop/git/wazuh/framework
plugins: metadata-1.10.0, html-2.1.1, tavern-1.7.0
collected 2 items                                                                                                                                                         

wazuh/tests/test_tasks.py ..                                                                                                                                        [100%]

============================================================================ 2 passed in 0.10s ============================================================================
```

## Integrated tests
```
adriiiprodri@wazuh pytest -x test_tasks_endpoints.tavern.yaml 
=========================================================================== test session starts ===========================================================================
platform linux -- Python 3.8.6, pytest-6.1.2, py-1.9.0, pluggy-0.13.1
rootdir: /home/adriiiprodri/Desktop/git/wazuh/api/test/integration, configfile: pytest.ini
plugins: metadata-1.10.0, html-2.1.1, tavern-1.7.0
collected 2 items                                                                                                                                                         

test_tasks_endpoints.tavern.yaml ..                                                                                                                                 [100%]

================================================================ 2 passed, 3 warnings in 91.78s (0:01:31) =================================================================

```

**Pull request closed without merging**